freqtrade/data/metrics.py

```diff
@@ -313,31 +317,64 @@ def calculate_sortino(
     return sortino_ratio
 
 
-def calculate_sharpe(
-    trades: pd.DataFrame, min_date: datetime, max_date: datetime, starting_balance: float
-) -> float:
+def calculate_account_value_history(
+    btdata: dict[str, DataFrame],
+    trades: DataFrame,
+    min_date: datetime,
+    max_date: datetime,
+    timeframe: str,
+    stake_currency: str,
+    start_balance: float,
+    pairlist: list[str],
+) -> Series:
+    """
+    Return a Series that contains the account value (approximately) at the open time of
+    every candle stick, in stake currency.
+    """
+    index = pd.date_range(min_date, max_date, freq=timeframe_to_resample_freq(timeframe))
+    df = pd.DataFrame(index=index)
+    df[stake_currency] = float(start_balance)
+    df[pairlist] = 0.0
+    for trade in trades.itertuples():
+        for order in trade.orders:
+            filled_at = pd.Timestamp(dt_from_ts(order["order_filled_timestamp"]))
+            stake = order["safe_price"] * order["amount"]
+
+            if order["ft_is_entry"]:
+                fee = stake * trade.fee_open
+            else:
+                fee = stake * trade.fee_close
+
+            if order["ft_order_side"] == "buy":
+                df.loc[filled_at:, trade.pair] += order["amount"]
+                df.loc[filled_at:, stake_currency] -= stake + fee
+            else:
+                df.loc[filled_at:, trade.pair] -= order["amount"]
+                df.loc[filled_at:, stake_currency] += stake - fee
+
+    for pair in pairlist:
+        close_price = btdata[pair].set_index("date")["close"]
+        df[pair] = df[pair] * close_price
+    value_in_stake_currency = df.sum(axis=1)
+    return value_in_stake_currency
+
+
+def calculate_sharpe(account_value_history: pd.Series, timeframe: str) -> float:
```

Comment:
while i don't necessarily mind having a "per account" sharpe ratio - it shouldn't hard-replace the existing one - at least not initially.